tsinfer/formats.py

```diff
@@ -1315,6 +1315,31 @@ def subset(self, individuals=None, sites=None, **kwargs):
             subset.record_provenance(command="subset", **kwargs)
         return subset
 
+    def get_sites_time_bound(self):
+        """
+        Returns a numpy array of the lower bound of the time of sites in the SampleData
+        file. Each individual with a nonzero time (from the individuals_time array)
+        gives a lower bound on the age of sites where the individual carries a
+        derived allele.
+
+        :return: A numpy array of the lower bound for each sites time.
+        :rtype: numpy.ndarray(dtype=uint32)
+        """
+        samples_individual = self.samples_individual[:]
+        samples_time = self.individuals_time[:][samples_individual]
+        if np.any(samples_time < 0):
+            raise ValueError("Individuals cannot have negative times")
+        historic_samples = samples_time != 0
+        historic_samples_time = samples_time[samples_time != 0]
+        sites_bound = np.zeros(self.num_sites)
+        for var in self.variants():
+            ancient_genos = var.genotypes[historic_samples]
+            if np.any(ancient_genos == 1):
```

Comment:
A bit more general to say ``ancient_genos > 0``? We might support non-biallelic *some* day.